src/compiler/checker.ts

```diff
@@ -16208,6 +16208,46 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                 /*readonly*/ false, target.labeledElementDeclarations && target.labeledElementDeclarations.slice(index, endIndex));
     }
 
+    function expandTupleShape(source: TupleTypeReference, target: TupleTypeReference): Type {
+        const sourceArity = getTypeReferenceArity(source);
+        const targetArity = getTypeReferenceArity(target);
+        const sourceElementTypes = getTypeArguments(source);
+        const targetElementTypes = getTypeArguments(target);
+
+        let resultTypes: Type[] | undefined;
+        let resultFlags: ElementFlags[] | undefined;
+
+        for (let i = 0; i < targetArity; i++) {
+            if (i >= sourceArity) {
+                if (!resultTypes) {
+                    resultTypes = sourceElementTypes.slice();
+                    resultFlags = source.target.elementFlags.slice();
+                }
+                const targetFlag = target.target.elementFlags[i];
+                resultTypes.push(anyType);

```

Comment:
Rather than `any` with the target flags, shouldn't it be `never` with the optional flag? To reflect how the consumer never reads the member?